### PR TITLE
[ROCm] Add GLM-5.3 k-pool compression and cache write

### PR DESCRIPTION
**Pull request by andyluo7**
## Summary

- add a fused GLM-5.3 k-pool compressor: per-dimension softmax pooling, BF16 boundaries, normalized Hadamard-128, and per-vector FP8 quantization
- write FP8 K values and FP32 scales into separate caller-owned paged caches using validated physical locations
- support an explicit write mask and reject out-of-range or duplicate active locations before launch
- select the platform FP8 E4M3 type from the target device instead of hard-coding the gfx942 or gfx950 spelling

## Cache contract

The production vLLM path uses one interleaved `uint8` allocation. TileLang tensors have one dtype, so this standalone example uses `[num_blocks, page_size, 128]` FP8 K storage plus `[num_blocks, page_size]` FP32 scale storage.

## Scope

This PR intentionally excludes decode-tail maintenance, pool Top-K, pool-to-token expansion, preshuffled framework layouts, and framework integration.

## Validation

Exact head: `213325ac1cc753855c58e19e74158778dff395ce`

- local `python3 -m py_compile` for all changed Python files
- local changed-file `pre-commit`: all hooks passed
- exact-head CI [run `35422594545`](https://github.com/tile-ai/tilelang/actions/runs/35422594545): Quick Lint, ROCm, CUDA, Metal, and CuTeDSL all passed
- ROCm 7.2 job [`105843032806`](https://github.com/tile-ai/tilelang/actions/runs/35422594545/job/105843032806): `2417 passed, 1850 skipped` on MI300X/gfx942
- independent MI350X/gfx950 run with an exact-head ROCm wheel and PyTorch 2.14.0+rocm7.2: all 4 focused k-pool tests passed

The unrounded-scale gfx942 result differs from the PyTorch reference at one of 640 values by exactly one adjacent E4M3FNUZ encoding. The test therefore checks finite values and a maximum distance of one FP8 ULP, while retaining a separate tight FP32 scale assertion.

## Summary

- Adds ROCm support for the GLM-5.3 fused K-pool compressor.
- Implements softmax pooling, BF16 boundaries, normalized Hadamard-128 rotation, and per-vector FP8 quantization.
- Writes FP8 K values and FP32 scales to separate paged caches.
- Adds location, mask, tensor, and cache validation.
- Adds device-aware ROCm FP8 type selection.
- Adds references, documentation, and focused tests.

## Scope

Excludes decode-tail maintenance, pool Top-K, pool-to-token expansion, preshuffled layouts, and framework integration.

## Validation

Validation includes Python compilation, pre-commit hooks, ROCm 7.2 MI300X CI, and focused MI350X tests. gfx942 checks allow one FP8 ULP for finite values and retain separate FP32 scale assertions.

## C++ style / lint notes

The PR does not change C++ code or C++ style documentation. The C++ API Style Audit is not relevant.

### CODE REVIEW

File: examples/kpool/example_glm53_kpool_compress.py [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+        if write_mask.dtype != torch.bool:
+            raise TypeError(f"write_mask must be torch.bool, got {write_mask.dtype}")
+        if write_mask.device != slot_k.device:
+            raise ValueError("write_mask must be on the same device as slot_k")
+        if not write_mask.is_contiguous():
+            raise ValueError("write_mask must be contiguous")
+
+    with torch.cuda.device(slot_k.device):
+        fp8_dtype_name = determine_fp8_type()
+        expected_fp8_dtype = determine_torch_fp8_type()
```
[the remark this answers is not in the document]
Comment (andyluo7) [reply examples/kpool/example_glm53_kpool_compress.py]: Fixed in efc39b98: both FP8 helpers now accept a target device, and the k-pool wrapper passes `slot_k.device`. Added a mocked heterogeneous-device regression covering gfx950 OCP selection on device 1 versus gfx942 FNUZ on device 0. The focused gfx942 k-pool suite passes all 4 cases.
